python/tk_desktop/desktop_engine_site_implementation.py

```diff
@@ -247,7 +247,8 @@ def run(self, splash, version, **kwargs):
         icon = QtGui.QIcon(":tk-desktop/default_systray_icon")
         app.setWindowIcon(icon)
 
-        splash.set_message("Building UI")
+        if splash:
```

Comment:
maybe add a comment here - and in the method docstring - pointing out that this can be none and when that happens. This feels like something that is super easy to miss when writing code - is there a way we could it differently? like a `set_splash()` method or something rather than having to do the none check every single time?